tests/Model/IndexInputTest.php

```diff
@@ -40,16 +42,22 @@ public function testConstructorShouldRequireNameToBeString(): void
         new IndexInput(['key' => ['x' => 1], 'name' => 1]);
     }
 
-    /** @dataProvider provideExpectedNameAndKey */
-    public function testNameGeneration($expectedName, array $key): void
+    /**
+     * @dataProvider provideExpectedNameAndKey
+     * @param array|object $key
+     */
+    public function testNameGeneration($expectedName, $key): void
     {
         $this->assertSame($expectedName, (string) new IndexInput(['key' => $key]));
     }
 
-    public function provideExpectedNameAndKey()
+    public function provideExpectedNameAndKey(): array
     {
         return [
             ['x_1', ['x' => 1]],
+            ['x_1', (object) ['x' => 1]],
+            ['x_1', new BSONDocument(['x' => 1])],
+            ['x_1', Document::fromPHP(['x' => 1])],
```

Comment:
I noted that the original tests for the function were more comprehensive (the four possible type casts are applied to each index specification), but I don't think there's any significant benefit to doing that. Testing a basic index spec with the non-array types seems sufficient.